drivers/SmartThings/zigbee-window-treatment/src/aqara/roller-shade/init.lua

```diff
@@ -0,0 +1,150 @@
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local FrameCtrl = require "st.zigbee.zcl.frame_ctrl"
+local data_types = require "st.zigbee.data_types"
+local aqara_utils = require "aqara/aqara_utils"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+
+local initializedStateWithGuide = capabilities["stse.initializedStateWithGuide"]
+local reverseRollerShadeDir = capabilities["stse.reverseRollerShadeDir"]
+local shadeRotateState = capabilities["stse.shadeRotateState"]
+local setRotateStateCommandName = "setRotateState"
+
+local MULTISTATE_CLUSTER_ID = 0x0013
+local MULTISTATE_ATTRIBUTE_ID = 0x0055
+local ROTATE_UP_VALUE = 0x0004
+local ROTATE_DOWN_VALUE = 0x0005
+
+
+local function window_shade_level_cmd(driver, device, command)
+  -- Cannot be controlled if not initialized
+  local initialized = device:get_latest_state("main", initializedStateWithGuide.ID,
+    initializedStateWithGuide.initializedStateWithGuide.NAME) or 0
+  if initialized == initializedStateWithGuide.initializedStateWithGuide.initialized.NAME then
+    aqara_utils.shade_level_cmd(driver, device, command)
+  end
+end
+
+local function window_shade_open_cmd(driver, device, command)
+  -- Cannot be controlled if not initialized
+  local initialized = device:get_latest_state("main", initializedStateWithGuide.ID,
+    initializedStateWithGuide.initializedStateWithGuide.NAME) or 0
+  if initialized == initializedStateWithGuide.initializedStateWithGuide.initialized.NAME then
+    device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, 100))
+  end
+end
+
+local function window_shade_close_cmd(driver, device, command)
+  -- Cannot be controlled if not initialized
+  local initialized = device:get_latest_state("main", initializedStateWithGuide.ID,
+    initializedStateWithGuide.initializedStateWithGuide.NAME) or 0
+  if initialized == initializedStateWithGuide.initializedStateWithGuide.initialized.NAME then
+    device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, 0))
+  end
+end
+
+local function write_tilt_attribute(device, payload)
+  local value = data_types.validate_or_build_type(payload, data_types.Uint16, "payload")
+  local message = cluster_base.write_attribute(device, data_types.ClusterId(MULTISTATE_CLUSTER_ID),
+    data_types.AttributeId(MULTISTATE_ATTRIBUTE_ID), value)
+  local frm_ctrl = FrameCtrl(0x10)
+  message.body.zcl_header.frame_ctrl:set_mfg_specific()
+  message.body.zcl_header.mfg_code = data_types.validate_or_build_type(aqara_utils.MFG_CODE, data_types.Uint16,
+    "mfg_code")
+  message.body.zcl_header.frame_ctrl = frm_ctrl

```

Comment:
all of this is equivalent to `cluster_base.write_manufacturer_specific_attribute`

Reply:
also you've immediately overwritten the result from calling `set_mfg_specific` with your own `FrameCtrl` object immediately afterwards